diesel/src/sqlite/connection/serialized_database.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(unsafe_code)]
+extern crate libsqlite3_sys as ffi;
+
+/// `SerializedDatabase` is a wrapper for a serialized database that is dynamically allocated by calling `sqlite3_serialize`.
+/// This RAII wrapper is necessary to deallocate the memory when it goes out of scope with `sqlite3_free`.
+#[derive(Debug)]
+pub struct SerializedDatabase {
```

Comment:
I think it might be meaningful to implement `Deref<Target = [u8]>` for this wrapper type so that it can be used as `&[u8]` directly.